prototype-spec.md

```diff
@@ -0,0 +1,253 @@
+# Notary v2 Specification
+
+This specification describes the file formats, queries, and processes required for the use of TUF in registries. It requires that metadata for each of the TUF roles: root, snapshot, timestamp, and targets is uploaded to a registry and can be queried by users. Details about these roles can be found in the [design document](https://github.com/notaryproject/nv2/blob/prototype-tuf/tuf-design.md) and the [TUF specification](https://github.com/theupdateframework/specification/blob/master/tuf-spec.md#the-update-framework-specification).
+
+## Root metadata
+
+Root metadata should be signed using an offline process. It can then be uploaded to the registry by running `nv2 upload root`.
+
+## Signing a manifest
+
+To upload a manifest signed with Notary v2, a developer must:
+* Generate and sign targets metadata by running `nv2 sign imagename keyloc --role targetsrolename` where `imagename` is the manifest they are signing, `targetsrolename` is the name of the top-level or delegated targets role the developer is a part of, and `keyloc` is a path to the developer's private key.
+* Upload the manifest and targets metadata to the registry by running `nv2 upload imagename`.
+* Send the targets metadata to the snapshot process. This may be part of the above command, but the developer should specify a `SNAPSHOT_LOCATION`.
+
+## Configuring delegations
+
+As developers join or leave teams, it may be necessary to update delegations. To do so, the delegator should perform the following steps:
+* To add a new delegatee they run `nv2 delegate fromrole torole --namespace ns --key keyloc`. This should add `torole` as a delegation to `fromrole` that is trusted for images in the namespace `ns`.
+* To remove a delegatee they run `nv2 revoke fromrole torole --key keyloc`.
+* After performing the above commands any number of times, they run `nv2 upload fromrole`. This will upload the new `fromrole` metadata. Any new delegated roles should sign and upload their targets metadata as well.
+
+
+## Snapshot and timestamp process
```

Comment:
Are snapshots always Merkle trees or not?

Reply:
Not always, Merkle trees wouldn't be needed for smaller organizations and the auditing requirement may be a challenge for these smaller organizations as well. It's mostly needed for folks like Docker that have thousands/millions of images. I did add a paragraph explicitly stating that Merkle trees are an option.